apps/E2E/src/RadioGroup/pages/RadioGroupPageObject.ts

```diff
@@ -24,52 +24,36 @@ class RadioGroupPage extends BasePage {
   /******************************************************************/
   /**************** UI Element Interaction Methods ******************/
   /******************************************************************/
-  // Get RadioButton's accessibilityLabel
-  async getRBAccessibilityLabel(radioButtonSelector: RadioButtonSelector): Promise<string> {
-    return await (await this.getRadioButton(radioButtonSelector)).getAttribute('Name');
-  }
 
   /* This resets the RadioGroup selection by clicking/selecting the 1st RadioButton in the RadioGroup.
    * Useful in beforeEach() hooks to reset the RadioGroup before additional tests */
   async resetRadioGroupSelection(): Promise<void> {
-    await (await this._firstRadioButton).click();
-  }
-
-  async getRadioButtonAccesibilityRole(): Promise<string> {
-    return await (await this._firstRadioButton).getAttribute('ControlType');
-  }
-
-  async isRadioButtonSelected(radioButtonSelector: RadioButtonSelector): Promise<boolean> {
-    return await (await this.getRadioButton(radioButtonSelector)).isSelected();
-  }
-
-  async clickRadioButton(radioButtonSelector: RadioButtonSelector): Promise<void> {
-    await (await this.getRadioButton(radioButtonSelector)).click();
+    await (await this.getRadioButton(RadioButton.First)).click();
   }
 
-  async waitForRadioButtonSelected(radioButtonSelector: RadioButtonSelector, timeout?: number): Promise<void> {
-    await browser.waitUntil(async () => await this.isRadioButtonSelected(radioButtonSelector), {
-      timeout: timeout ?? this.waitForUiEvent,
-      timeoutMsg: 'RadioButton was not selected correctly.',
-      interval: 1000,
-    });
+  async isRadioButtonSelected(selector: RadioButton): Promise<boolean> {
+    return await (await this.getRadioButton(selector)).isSelected();
   }
 
-  /* Sends a Keyboarding command on a specific UI element */
-  async sendKey(key: string, radioButtonSelector: RadioButtonSelector): Promise<void> {
-    await (await this.getRadioButton(radioButtonSelector)).addValue(key);
+  async waitForRadioButtonSelected(selector: RadioButton, errorMsg?: string): Promise<boolean> {
+    await this.waitForCondition(
+      async () => await this.isRadioButtonSelected(selector),
+      errorMsg ?? `RadioButton ${selector} failed to be selected.`,
```

Comment:
Let's make `errorMsg` a mandatory parameter here. We want to make sure the user passes in their specific scenario here